workers/data_refinery_workers/processors/illumina.py

```diff
@@ -0,0 +1,259 @@
+from __future__ import absolute_import, unicode_literals
+
+import csv
+import os
+import string
+import subprocess
+import multiprocessing
+import warnings
+from django.utils import timezone
+from typing import Dict
+
+import rpy2.robjects as ro
+from rpy2.rinterface import RRuntimeError
+
+from data_refinery_common.logging import get_and_configure_logger
+from data_refinery_common.models import (
+    OriginalFile, 
+    ComputationalResult, 
+    ComputedFile,
+    Sample,
+    OriginalFileSampleAssociation,
+    SampleResultAssociation
+)
+from data_refinery_workers._version import __version__
+from data_refinery_workers.processors import utils
+from data_refinery_common.utils import get_env_variable
+S3_BUCKET_NAME = get_env_variable("S3_BUCKET_NAME", "data-refinery")
+
+logger = get_and_configure_logger(__name__)
+
+
+def _prepare_files(job_context: Dict) -> Dict:
+    """Adds the keys "input_file_path" and "output_file_path" to
+    job_context so everything is prepared for processing.
+    """
+    original_file = job_context["original_files"][0]
+    job_context["input_file_path"] = original_file.absolute_file_path
+    # Turns /home/user/data_store/E-GEOD-8607/raw/foo.txt into /home/user/data_store/E-GEOD-8607/processed/foo.cel
+    pre_part = original_file.absolute_file_path.split('/')[:-2]
+    end_part = original_file.absolute_file_path.split('/')[-1]
+    job_context["output_file_path"] = '/'.join(pre_part) + '/processed/' + end_part
+    job_context["output_file_path"] = job_context["output_file_path"].replace('.txt', '.PCL')
+
+    # Sanitize this file so R doesn't choke.
+    # Some have comments, some have non-comment-comments.
+    file_input = open(job_context["input_file_path"], "r")
+    with open(job_context["input_file_path"], "r") as file_input:
+        with open(job_context["input_file_path"] + ".sanitized", "w") as file_output:
+            for line in file_input:
+                if '#' not in line and \
+                line != None and \
```

Comment:
I'm not sure if `line` will ever actually be `None` or not, but this condition will never be reached because if `line` is `None` then `'#' not in line` will raise an exception.